core/metrics/src/pusher.rs

```diff
@@ -1,12 +1,17 @@
 use awc::error::{ConnectError, SendRequestError};
 use awc::Client;
+use lazy_static::lazy_static;
+use percent_encoding::{utf8_percent_encode, AsciiSet, NON_ALPHANUMERIC};
 use tokio::time::{self, Duration, Instant};
 
-use percent_encoding::{utf8_percent_encode, NON_ALPHANUMERIC};
 use ya_core_model::identity::{self, IdentityInfo};
 use ya_service_api::MetricsCtx;
 use ya_service_bus::typed as bus;
 
+lazy_static! {
+    static ref SAFE_CHAR_SET: AsciiSet = NON_ALPHANUMERIC.remove(b'.').remove(b'-').remove(b'_');
```

Comment:
This characters are rather an `UNSAFE_CHAR_SET`, because you want to encode them

Reply:
Changed